README.md

```diff
@@ -1,32 +1,289 @@
+| Modifying each element and returning the modified array        | `map()`               |
+| Creating a summary or aggregation of values in an array        | `reduce()`            |
 
-## Topic 1
+In this lesson, we will learn how to use the first of the iterator methods
+above: `forEach()`. We will also learn under what circumstances `forEach()` is
+the right iterator method to use. But first, we're going to get a little more
+context for these methods by learning more about JavaScript objects.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+## A Deeper Look at Objects
 
-## Topic 2
+We've seen that JavaScript has many built-in methods that we can use to make our
+code more efficient and to avoid having to "reinvent the wheel." But what
+exactly is a method? A method is nothing more than a function that belongs to an
+Object. So what does that mean? Let's take a look at an example.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+Let's create a `dog` object and give it a couple of properties:
+
+```js
+const dog = {
+    name: "Spinach",
+    breed: "Shiba Inu"
+}
+```
+
+As you can see, this object represents a particular dog:
+
+```js
+dog.name;
+// => "Spinach"
+dog.breed;
+// => "Shiba Inu"
+```
+
+In addition to having the `name` and `breed` properties, we might also want our
+`dog` object to be able to `bark()`:
+
+```js
+function bark() {
+    console.log("Woof woof!")
+}
+```
+
+Now we can call our `bark()` function, but it isn't associated with our `dog`
+object in any way. It would really make more sense for the `bark()` function to
+_belong_ to our `dog` object somehow. Well, we can do that!
+
+You may recall from an earlier lesson that, in JavaScript, functions are
+considered _first class_. This means that, like data values, they can be
+assigned to variables and used as arguments in other functions. It also means
+you can store functions in Arrays and Objects. In order to make the `bark()`
+function belong to our `dog`, we can simply add it to the `dog` object as
+another property:
+
+```js
+const dog = {
+    name: "Spinach",
+    breed: "Shiba Inu",
+    bark: function() {
+        console.log("Woof woof!");
+    }
+}
+```
+
+Now that `bark()` belongs to an object, we refer to it as a `method` rather than
+a `property`. A method in JavaScript is simply a function that belongs to an
+object.
+
+<details><summary><b>How do you think we would call the `bark()` method?</b></summary><p>
+    <code>dog.bark();</code>
+    <p>We call a method the same way we access a property: using the dot operator. If we just enter <code>dog.bark</code>, without the parentheses, the method itself will be returned:</p>
+    <code>dog.bark</code><br/>
```

Comment:
Maybe wrap these lines in just one <code> to make it all one block? 

```
<code>
        dog.bark<br/>
        // => f () {<br/>
        // console.log("Woof woof!");<br/>
        // }
</code>
```

It worked in my VSCode MD preview, but unsure if it will actually work in practice. Still maybe worth a shot. If it works, same applies to the other code blocks within details throughout this lesson